lib/rspec/rails/adapters.rb

```diff
@@ -60,6 +60,48 @@ def assertion_instance
       end
     end
 
+    # @private
+    class ControllerAssertionDelegator < AssertionDelegator
+      module Setup
+        extend ActiveSupport::Concern
+
+        included { init_setup }
+
+        module ClassMethods
+          attr_reader :setup_methods, :setup_blocks
+          attr_accessor :controller_class
+
+          def init_setup
+            @setup_methods ||= []
+            @setup_blocks ||= []
+          end
+
+          def setup(*methods, &block)
+            @setup_methods += methods
+            @setup_blocks << block if block
+          end
```

Comment:
You may have found it yourself already, but anyway - if you add a `def teardown(*); end` here, the spec starts failing with a different error.